### DISCUSSION TITLE
Overriding "underlying" providers

### DISCUSSION THREAD

**Discussion by snesbittsea** [ets-labs/python-dependency-injector#500] [Q&A]
The documentation contains an example of a passing arguments to underlying providers as follows:
```
    algorithm_factory = providers.Factory(
        Algorithm,
        task=providers.Factory(
            ClassificationTask,
            loss=providers.Factory(
                Loss,
                regularizer=providers.Factory(
                    Regularizer,
                ),
            ),
        ),
    )```

My question is in this situation how does one override an underlying provider? For example, using a mock for the loss provider?

**Comment by hakanutku** [comment]
`containers.py`
```python
regularizer = providers.Factory(Regularizer)
loss = providers.Factory(Loss, regularizer=regularizer)
task = providers.Factory(ClassificationTask, loss=loss)

algorithm_factory = providers.Factory(Algorithm, task=task)
```
`main.py`
```python
with container.regularizer.override(mock_regularizer):
    algorithm = container.algorithm_factory() # Algorithm with mock regularizer
```

I think this should work.
